apps/heft/src/utilities/resolvePackage.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { sync } from 'resolve';
+import * as path from 'path';
+import { FileSystem, PackageJsonLookup } from '@rushstack/node-core-library';
+
+const packageJsonLookup: PackageJsonLookup = new PackageJsonLookup();
+
+/**
+ * Options for package path resolution
+ * @public
+ */
+export interface IResolvePackageOptions {
+  doNotResolveSymlinks: boolean;
+}
+
+/**
+ * Gets the path of a given package name
+ * @public
+ */
+export function resolvePackage(
+  packageName: string,
+  rootPath: string,
+  options: Partial<IResolvePackageOptions> = {}
+): string {
+  options = {
+    doNotResolveSymlinks: false,
+    ...options
+  };
+
+  let normalizedRootPath: string = options.doNotResolveSymlinks ? rootPath : FileSystem.getRealPath(rootPath);
+  normalizedRootPath = packageJsonLookup.tryGetPackageFolderFor(normalizedRootPath) || normalizedRootPath;
+
+  try {
+    return path.dirname(
+      sync(packageName, {
+        basedir: normalizedRootPath,
+        packageFilter: (pkg: { main: string }): { main: string } => {
+          pkg.main = 'package.json';
+          return pkg;
+        }
+      })
+    );
+  } catch (e1) {
+    try {
+      // If we fail, see if we're trying to resolve to the current package
+      const currentPackageJson: { name: string } = require(path.join(normalizedRootPath, 'package.json'));
+      if (currentPackageJson.name === packageName) {
+        return normalizedRootPath;
+      } else {
+        throw e1;
+      }
+    } catch (e2) {
+      throw e1;
+    }
+  }
+}
+
+/**
+ * Resolves a path in a package, relative to another path.
+ */
+export function resolvePackagePath(
```

Comment:
Why aren't we using the [resolve](https://www.npmjs.com/package/resolve) library? This seems like a complex wheel to reinvent. (?)

Reply:
This does use the `resolve` library. See line 4.